kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -313,132 +341,140 @@ define([
+                return params.specs[id];
+            });
+        }
+
+        function createFilePathWidget(appSpec, filePathParams, parameterId) {
+            const spec = filePathParams.paramMap[parameterId];
+            return Promise.try(() => {
+                makeFieldWidget(appSpec, spec, initialParams[spec.id], filePathParams.layout)
+                    .then((widget) => {
+                        widgets.push(widget);
+                        return widget.start({
+                            node: document.getElementById(filePathParams.view[spec.id].id)
+                        });
                     });
+            }).catch((ex) => {
+                const errorDisplay = div({
+                    class: 'kb-field-widget__error_message--file-paths'
+                }, [
+                    ex.message
+                ]);
+                document.getElementById(filePathParams.view[spec.id].id).innerHTML = errorDisplay;
+
+                throw new Error('Error making input field widget', ex);
             });
+
+        }
+
+        function deleteRow(e){
+            $(e.target).closest('tr').remove();
+            let filePathRows = ui.getElements(`${cssClassType}-fields-row`);
+            updateRowNumbers(filePathRows);
+        }
+
+
+        function renderFilePathRow(filePathRow) {
+            const appSpec = model.getItem('appSpec');
+
+            const params = model.getItem('parameters');
+            let filePathParams = makeFilePathsLayout(findPathParams(params));
+
+            if (!filePathParams.layout.length) {
+                ui.getElement(`${cssClassType}s-area`).classList.add('hidden');
+            } else {
+                filePathRow.innerHTML = div({
+                    class: `${cssBaseClass}__param_container row`,
+                }, [
+                    filePathParams.content
+                ]);
+
+                $(filePathRow).prepend(
+                    td({
+                        class: `${cssBaseClass}__file_number`,
+                    })
+                );
+
+                $(filePathRow).append(
+                    td({}, [
+                        button({
+                            class: 'btn btn__text',
+                            type: 'button',
+                            id: events.addEvent({
+                                type: 'click',
+                                handler: function(e){
+                                    deleteRow(e);
+                                }
+
+                            })
+                        },[
+                            icon({
+                                class: 'fa fa-trash-o fa-lg',
+                            })
+                        ])
+                    ])
+                );
+
+                Promise.all(filePathParams.layout.map((parameterId) => {
+                    createFilePathWidget(appSpec, filePathParams, parameterId);
+                }));
+
+                events.attachEvents(container);
```

Comment:
should the call to attach events happen after the promise.all call resolves or is it ok to do it async? if the former then you'll want to add a `then()` handler to the promise and shove this call in there